src/front-end/typescript/lib/pages/opportunity/code-with-us/edit/tab/history.tsx

```diff
@@ -61,8 +242,94 @@ const view: ComponentView<State, Msg> = ({ state, dispatch }) => {
   );
 };
 
+//props for the AttachmentsView component
+interface Props extends ComponentViewProps<State, Msg> {
+  disabled?: boolean;
+}
+
+// @duplicated-attachments-view
+const AttachmentsView: View<Props> = ({ state, dispatch, disabled }) => {
+  return (
+    <Row>
+      <Col xs='12'>
+        <Attachments.view
+          dispatch={mapComponentDispatch(dispatch, msg => adt('attachments' as const, msg))}
+          state={state.attachments}
+          disabled={disabled}
+          // className='mt-4'
+          />
+      </Col>
+    </Row>
+  );
+};
+
 export const component: Tab.Component<State, Msg> = {
   init,
   update,
-  view
+  view,
+  getModal: state => {
+    if (!state.showModal) { return null; }
+    switch (state.showModal.tag) {
+      case 'addNote': {
+        return {
+          title: 'Add Entry to Opportunity',
+          onCloseMsg: adt('hideModal'),
+          body: dispatch => {
+          const attachmentProps = { state, dispatch, disabled: false }
+
+
+            return (
+              <div>
+                <p>Provide a note or commentary below pertaining to the opportunity. You may also include any applicable attachments.</p>
+                <Alert color='danger' style={{ whiteSpace: 'pre-line' }}><strong>Important! </strong>The notes and any attachments, if applicable, cannot be edited or deleted once submitted.
+                </Alert>
+                <LongText.view
+                  extraChildProps={{style: { height: '125px' }}}
+                  label='Notes'
+                  required
+                  state={state.modalNote}
+                  dispatch={mapComponentDispatch(dispatch, value => adt('modalNote' as const, value))} />
+              <AttachmentsView {...attachmentProps} />
+              </div>
+
+            );
+          },
+          actions: [
+            {
+              text: 'Submit Entry',
+              button: true,
+              disabled: state.modalNote.child.value.trim().length === 0 || state.modalNote.child.value.length > 2000,
```

Comment:
Is this enough of a validation for the note length? In the DB, note is a [text type](https://www.postgresql.org/docs/9.1/datatype-character.html), which has no limit--should I change it to something like varchar? If yes, do I need a new migration, or can it be rolled into the one already in this PR?